stack/global/cloudtrail.tf

```diff
@@ -1,20 +1,22 @@
 data "aws_caller_identity" "current" {}
 
 module "log-producer" {
+  count  = var.cloudtrail_config == null ? 0 : 1
   source = "../../cloudtrail/log-producer"
 
   environment                        = var.environment
   project                            = var.project
   is_organization_trail              = false
   is_multi_region_trail              = true
   enable_management_cloudtrail       = true
-  enable_data_cloudtrail             = var.enable_data_cloudtrail
-  s3_bucket_arns_for_data_cloudtrail = var.s3_bucket_arns_for_data_cloudtrail
-  cloudtrail_target_bucket_name      = module.log-ingestor.s3_bucket_name
-  cloudtrail_target_bucket_kms_arn   = module.log-ingestor.s3_kms_arn
+  enable_data_cloudtrail             = var.cloudtrail_config.enable_data_cloudtrail
+  s3_bucket_arns_for_data_cloudtrail = var.cloudtrail_config.s3_bucket_arns_for_data_cloudtrail
+  cloudtrail_target_bucket_name      = module.log-ingestor[0].s3_bucket_name
```

Comment:
What if this is a 2nd account, that only produces logs? The logs are then ingested by another account, meaning we'd have to pass the target bucket name in as a variable?

I think we want to either pass in an existing ingestor account's information, or we create the ingestor here.